src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,160 @@
+
+Before diving into the specification with all the edge cases, let's look at two simple examples:
+
+1. The user clicks a button that triggers some requests to the backend and stores data in the
+local database. The SDK can create a meaningful transaction in that case.
+2. The user clicks a button that solely validates some form data and triggers nothing that the
+SDK can automatically instrument. The SDK could only create a transaction without any spans, which would
+be valueless.
+
+# Specification
+
+Users can change the `idleTimeout` via the SDK config options. The default is 3.0 seconds. 
+
+The specification is written in the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+
+```Gherkin
+Scenario: Starting UI Event transactions
+    Given an instrumentable UI event
+    Then the SDK starts a UI event transaction 
+    And starts the idle timeout with idleTimeout of the options
+
+Scenario: Wait for children when starting span
+    Given a UI event transaction
+    When the SDK starts a child span
+    Then the SDK cancels the idle timeout
+    And waits for the child to finish
+
+Scenario: Start timeout when the last span finishes
+    Given a UI event transaction
+    And the transaction has one or multiple running child spans
+    And the transaction is waiting for its children to finish
+    When the SDK finishes the last child span
+    Then the SDK starts the idle timeout
+
+Scenario: Don't reset timeout when the second last span finishes
+    Given a UI event transaction
+    And the transaction has two running child spans
+    When the SDK finishes the first child span
+    Then the SDK doesn't reset the idle timeout
+    And waits for the children to finish
+
+Scenario: Discard UI event transactions without child spans
+    Given a UI event transaction
+    And the transaction has no child spans
+    When the idleTimeout times out
+    Then the SDK discards the transaction 
+
+Scenario: Set time to last finished child span
+    Given a UI event transaction
+    And the transaction has one finished child span
+    And the transaction has one running child span
+    When the running child span finishes
+    Then the SDK finishes the transaction
+    And trims the end time of the transaction to the one of the last finished
+        child span
+
+Scenario: Same UI element with same event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI event with the same type for the same
+        UI element
+    Then the SDK resets the timeout
+    And doesn't create a new transaction
+
+# If your SDK binds auto-genrated transactions to the scope see binding
+# to scope
+Scenario: Same UI element with different event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI element with a different event
+    Or the user triggers a different UI element
+    Then the SDK finishes the ongoing transaction
```

Comment:
`h`: I think it's important to mention here that we also remove it from scope (on Android we do), because calling `finish` doesn't necessarily trigger the immediate finish due to `waitForChildren`

```suggestion
    Then the SDK finishes the ongoing transaction
    And removes the transaction from scope
```

Reply:
ahh, just noticed it's in the different section, sorry my bad!